### PR TITLE
[AF-837] Add telemetry mechanism

### PR DESCRIPTION
**Pull request by svizzari**
/cc @zendesk/vegemite

### Description
It would be very helpful if we could have the migrator "call home" periodically. This could be an opt-in feature, i.e. we'd ask the user whether they would allow us to send anonymous data from time to time, so as to improve the cli-tool.

screenshot 2017-10-24 13 59 34

### Tasks
- [x] Replace [placeholder tracking code](https://github.com/zendesk/zendesk_app_migrator/pull/29/files#diff-ad53372f51e33a9eea8e56eaa7722408R69) once we have a GA property
- [x] Write tests

### References
* JIRA: https://zendesk.atlassian.net/browse/AF-837

### Risks
* N/A

### CODE REVIEW

**Comment by Mehul-Porwal** [comment]
This is not capturing `Number of times the migrator was run per-app` or is it?

**Comment by svizzari** [comment]
@Mehul-Porwal 

> This is not capturing Number of times the migrator was run per-app or is it?

No, I opted to make anonymity more of a concern.  This iteration just tracks basic interactions.

**Comment by Mehul-Porwal** [comment]
Can you update the JIRA or add that on the JIRA as a comment? The JIRA task expects this metric to be tracked.

**Comment by svizzari** [comment]
@Mehul-Porwal 💇 comments addressed

File: src/test/migrator.test.ts [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+        expect(askStub.called).to.be.false;
+      });
+      it("shouldn't try to track data", async () => {
+        const _saveStub = stub(mgtr.insight, "_save");
+        await Migrator.migrate(
+          merge(cliOptions, {
+            noInsight: true
+          })
+        );
+        expect(_saveStub.called).to.be.false;
```
Comment (hoangtrvu) [inline src/test/migrator.test.ts]: can we use `trackStub` here?

Comment (svizzari) [reply src/test/migrator.test.ts]: Nope, the reason for this is that `track` is still called, but [short-circuits](https://github.com/yeoman/insight/blob/master/lib/index.js#L94-L96) based on the value of the `optOut` property.

**Pull request merged**